From 31646ee20b6dd8ce07a5c2f7e4583a20a0486146 Mon Sep 17 00:00:00 2001
From: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Date: Wed, 13 Jul 2022 19:47:18 +0530
Subject: [PATCH] Fix bluetooth crash when stack unable to start HCI_MODULE

When BT is in Host controlled mode for CIV, if user tries
to turn ON, Bluetooth crashes continously. Added fix in
stack to gracefully shutdown HCI_MODULE if it is unable to
start it.

Change-Id: I98d3df271c9c7d97d78e7c426414b65c1946f9fc
Tracked-On: OAM-89874
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Co-authored-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 system/btcore/src/module.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/system/btcore/src/module.cc b/system/btcore/src/module.cc
index f0d643d63f..8df9defceb 100644
--- a/system/btcore/src/module.cc
+++ b/system/btcore/src/module.cc
@@ -100,7 +100,7 @@ void module_shut_down(const module_t* module) {
   CHECK(state <= MODULE_STATE_STARTED);
 
   // Only something to do if the module was actually started
-  if (state < MODULE_STATE_STARTED) return;
+  if (state < MODULE_STATE_STARTED && strcmp(module->name, "hci_module")) return;
 
   LOG_INFO("%s Shutting down module \"%s\"", __func__, module->name);
   if (!call_lifecycle_function(module->shut_down)) {
-- 
2.17.1

